src/app/tags/views/TagMachines/TagMachines.tsx

```diff
@@ -1,38 +1,61 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 
+import type { ValueOf } from "@canonical/react-components";
 import { Spinner } from "@canonical/react-components";
 import { useDispatch, useSelector } from "react-redux";
 
+import ErrorsNotification from "../../../machines/views/MachineList/ErrorsNotification";
+
 import ModelNotFound from "app/base/components/ModelNotFound";
 import { useWindowTitle } from "app/base/hooks";
 import { useGetURLId } from "app/base/hooks/urls";
+import type { SortDirection } from "app/base/types";
 import urls from "app/base/urls";
 import MachineListTable, {
   DEFAULTS,
 } from "app/machines/views/MachineList/MachineListTable";
-import machineSelectors from "app/store/machine/selectors";
+import type { FetchGroupKey, FetchFilters } from "app/store/machine/types";
+import { mapSortDirection } from "app/store/machine/utils";
 import { useFetchMachines } from "app/store/machine/utils/hooks";
 import type { RootState } from "app/store/root/types";
 import { actions as tagActions } from "app/store/tag";
 import tagSelectors from "app/store/tag/selectors";
 import { TagMeta } from "app/store/tag/types";
+import { FetchNodeStatus } from "app/store/types/node";
 import { isId } from "app/utils";
 
 export enum Label {
   Machines = "Deployed machines",
 }
 
+const PAGE_SIZE = DEFAULTS.pageSize;
+
 const TagMachines = (): JSX.Element => {
   const dispatch = useDispatch();
   const id = useGetURLId(TagMeta.PK);
   const tag = useSelector((state: RootState) =>
     tagSelectors.getById(state, id)
   );
   const tagsLoading = useSelector(tagSelectors.loading);
-  const deployedMachines = useSelector((state: RootState) =>
-    machineSelectors.getDeployedWithTag(state, id)
+  const [currentPage, setCurrentPage] = useState(1);
+  const [sortKey, setSortKey] = useState<FetchGroupKey | null>(
+    DEFAULTS.sortKey
   );
-  useFetchMachines();
+  const [sortDirection, setSortDirection] = useState<
+    ValueOf<typeof SortDirection>
+  >(DEFAULTS.sortDirection);
+  let filters: FetchFilters = { status: FetchNodeStatus.DEPLOYED };
+  if (tag) {
+    filters.tags = [tag.name];
+  }
+  const { callId, loading, machineCount, machines, machinesErrors } =
+    useFetchMachines({
+      currentPage,
+      filters,
+      pageSize: PAGE_SIZE,
+      sortDirection: mapSortDirection(sortDirection),
```

Comment:
`useFetchMachines` hook should handle value mapping by default IMO.

Reply:
I miiiight leave this until we see how the hook gets used in the DHCP and clone forms etc. as this might tie the hook to the table sort but we might not use that in all places (or maybe we will).

I've added an issue to do the update once we've taken a look at the forms: https://github.com/canonical/app-tribe/issues/1320.